src/Microsoft.VisualStudio.Threading/TplExtensions.cs

```diff
@@ -108,6 +108,26 @@ public static async Task<T> WithTimeout<T>(this Task<T> task, TimeSpan timeout)
             return task.GetAwaiter().GetResult();
         }
 
+        /// <summary>
+        /// Gets the <see cref="Task{TResult}.Result"/> of a completed <see cref="Task{TResult}"/>.
+        /// </summary>
+        /// <typeparam name="TResult">The type of the result produced by the <see cref="Task{TResult}"/>.</typeparam>
+        /// <param name="task">The completed task.</param>
+        /// <returns>The result value of the <see cref="Task{TResult}"/>.</returns>
+        /// <exception cref="ArgumentNullException">If <paramref name="task"/> is <see langword="null"/>.</exception>
+        /// <exception cref="InvalidOperationException">If <paramref name="task"/> is not completed.</exception>
+        public static TResult CompletedResult<TResult>(this Task<TResult> task)
+        {
+            Requires.NotNull(task, nameof(task));
+
+            if (!task.IsCompleted)
+            {
+                throw new InvalidOperationException();
+            }
+
+            return task.Result;
```

Comment:
Should we use `GetAwaiter().GetResult()` here instead so that rethrown exceptions have the expected type the way *await* does?

Reply:
➡️ No. This is a replacement for `Task<TResult>.Result` in contexts where the result is known to be completed. It should preserve both the success and error behaviors from that property.